### ISSUE TITLE
docs: showcase the superquadric shape family and differentiable fitting

### ISSUE THREAD

**Issue by KhoiDOO** [KhoiDOO/conquer3d#46]
`conquer3d.primitive` has no figure on the showcase. Superquadrics are the library's only parametric primitive family and its only learnable `nn.Module`, and neither the shape family they span nor the differentiable fitting built on them is shown anywhere.

Two cards are wanted:

1. **The shape family.** A superquadric is controlled by two exponents, and the interesting fact about the family is that those two numbers sweep continuously from a box through an ellipsoid to a pinched star. A lattice over both exponents shows that in one image, and nothing in the docs currently does.

2. **Differentiable fitting.** `examples/differentiable_fitting/superquadrics.py` fits a learnable set to a mesh with the SuperFlex objective, but only ever at its default of 24 primitives, and only as a script that writes PLY files. Fitting at scale against a mesh with thin features -- the Horse -- would show both the abstraction and the reconstruction.

Separately, the landing page opens with a "What it is" section whose four feature cards restate the hero paragraph and its code sample directly above them, so the page says the same thing twice before showing a single figure.

**Comment by KhoiDOO** [comment]
Landed on `docs/superquadric-showcases` in two commits.

## API usage

**Shape family.** All 64 primitives are built in one call and sliced apart, rather than 64 separate sets:

```python
import torch
from conquer3d.primitive import SuperQuadrics
from conquer3d.primitive.sq import MIN_EXPONENT, MAX_EXPONENT   # 0.1, 1.9

axis = torch.linspace(MIN_EXPONENT, MAX_EXPONENT, 8, device="cuda")
e1 = axis.repeat_interleave(8)      # down the rows
e2 = axis.repeat(8)                 # across the columns

quats = torch.zeros(64, 4, device="cuda"); quats[:, 0] = 1.0
sq = SuperQuadrics.from_values(
    scales=torch.ones(64, 3, device="cuda"),
    exponents=torch.stack([e1, e2], dim=-1),
    quaternions=quats,
    translations=torch.zeros(64, 3, device="cuda"),
    learnable=False,
)

verts, faces = sq.get_mesh(resolution=64)     # (V, 3) float32, (F, 3) int32
# get_mesh concatenates in order with a fixed count per primitive, so primitive
# i is a slice, not a label lookup:
v_per, f_per = 64 * 62 + 2, 2 * 64 * 62
vi = verts[i * v_per:(i + 1) * v_per]
fi = faces[i * f_per:(i + 1) * f_per] - i * v_per
```

**Fitting**, and the part that matters — reading the result back as a surface rather than as a pile of primitives:

```python
from conquer3d.data_structure import create_voxel_grid
from conquer3d.ops import marching_cubes
from conquer3d.primitive import compute_sq_union

@torch.no_grad()
def union_surface(sq, res=160, chunk=8192):
    """The boundary of the union solid. Chunked because the field is (K, M) wide."""
    grid_vertices, voxels, _ = create_voxel_grid(
        grid_min=[-0.62] * 3, grid_max=[0.62] * 3, res=[res] * 3, device="cuda")
    field = torch.empty(grid_vertices.shape[0], device="cuda")
    for i in range(0, grid_vertices.shape[0], chunk):
        block = grid_vertices[i:i + chunk].contiguous()
        field[i:i + chunk] = compute_sq_union(sq(block), tau=sq.union_tau, mask=sq.mask)
    return marching_cubes(grid_vertices, voxels, field, iso=0.0)[:2]
```

## Advantages demonstrated

- **The whole shape family is two numbers.** One 8x8 lattice covers box, rounded box, cylinder-ish, ellipsoid, sphere and pinched star, with no special cases in the tessellator.
- **Fitting at 2,000 primitives is routine**, 83x the example's default of 24: 800 Adam steps in **167 s**, soft IoU **0.993**, union Chamfer **0.094 -> 0.008** against the Horse (96,966 faces).
- **The union closes into a watertight mesh.** `compute_sq_union` on a 160^3 grid plus `marching_cubes` turns 2,000 overlapping primitives into a single 61,012-face surface — the two halves of the library meeting.

## Disadvantages and known trade-offs

- **K * M is a hard budget.** `compute_sq_sdf` materialises the full `(K, M)` field and keeps ~20 intermediates for autograd. Measured at **230 bytes per element**, so a 24 GB card tops out at `K * M ~ 1.0e8`. At the example's M = 104,096 supervision points, K = 768 takes 17.4 GiB and K = 1,000 OOMs. K = 100,000 would need 41 GB for a single tensor and ~42 s/step.
- **Chunking does not lift that ceiling for training.** The softmin union needs all K values at once, so reaching six figures needs a custom chunked-logsumexp backward that recomputes chunks — a library feature, not a caller-side fix. Inference is fine, because `no_grad` lets the field be streamed (that is what `union_surface` above does).
- **`get_mesh` is a concatenation, not a union.** It keeps one closed surface per primitive, including surface buried inside neighbours, and its Euler characteristic is `2K`. It is the right output for showing a decomposition and the wrong one for measuring a fit.
- **Fitting favours volume over surface.** The objective is dominated by a volumetric IoU term; primitives grow to fill the interior, and surface fidelity is carried only by the truncated SDF term. Soft IoU reaches 0.993 while Chamfer is still 0.008 — the metrics saturate at different times.

## Empirical results

Horse, 96,966 faces, normalised to a half-extent of 0.5. 800 Adam steps, cosine-annealed, on an RTX 4090. Chamfer measured on the **union surface** against 200,000 area-weighted samples of the target:

| K | M (occupancy + surface) | K*M | soft IoU | union Chamfer | fit time |
| ---: | ---: | ---: | ---: | ---: | ---: |
| 10,000 | 7,000 | 7.0e7 | 0.9966 | 0.057 -> 0.035 | 166 s |
| 2,000 | 35,000 | 7.0e7 | 0.9933 | 0.094 -> **0.008** | 167 s |

Same budget, same wall-clock, **4x** the accuracy from spending it on supervision instead of primitives. The shipped figure uses K = 2,000.

Memory ceiling, measured directly (`K * M` is the only variable that matters — peak is identical at 8.74 GiB for every row of the second group):

| K | M | K*M | peak | result |
| ---: | ---: | ---: | ---: | :--- |
| 128 | 104,096 | 1.3e7 | 2.91 GiB | ok |
| 512 | 104,096 | 5.3e7 | 11.66 GiB | ok |
| 768 | 104,096 | 8.0e7 | 17.44 GiB | ok |
| 1,000 | 104,096 | 1.0e8 | — | OOM |
| 2,000 / 5,000 / 10,000 / 20,000 / 50,000 | 20,000 / 8,000 / 4,000 / 2,000 / 800 | 4.0e7 | 8.74 GiB | ok |

Shape family: 64 primitives at angular resolution 64, 3,970 vertices and 7,936 faces each, tessellated and rendered in **7.7 s**.

## Technical insights worth preserving

- **Chamfer on `get_mesh` output inverts the conclusion.** Sampling the concatenation hits surface interior to neighbouring primitives, and that interior grows as primitives grow — so every configuration scored *worse* than its own k-means initialisation (0.017 -> 0.070 at K = 10,000). Measuring the union boundary instead turns the identical run into 0.094 -> 0.008. `sq.py`'s `get_mesh` note already warns about this; it is worth heeding.
- **The softmin bias scales with `log K`, and it bites.** `compute_sq_union` is a lower bound short of the true minimum by `tau * log(K)`. At the default `tau = 0.01`, that is 0.076 at K = 2,000 and 0.092 at K = 10,000 — a seventh to a fifth of the object's half-extent. The union inflates, the fit scatters trying to compensate, and the result is visibly wrong. `tau = 1e-3` fixes it. **`union_tau` should arguably default to something K-dependent** rather than a constant tuned at K = 24.
- **The overlap penalty is inert at scale.** Disabling it entirely at K = 10,000 changed the result not at all — soft IoU and Chamfer were bit-identical. It is tuned for tens of primitives, where stacking is the failure mode; at thousands it contributes nothing measurable.
- **k-means initialisation must be vectorised.** The example loops over centres, which is fine at 24 and hopeless at thousands. Scattering every point to its centre with `index_add_` is ~4 lines and runs at any K; it also needs more sample points than centres to mean anything, so the init cloud is sized at `max(4K, 40,000)`.
- **The Horse asset is Y-long and Z-up** against a Y-up renderer, so untransformed it stands on its nose. `(x, y, z) -> (x, z, -y)` has determinant +1 — a rotation, not the reflection `(x, z, y)` would give.
- **A 64-cell parameter sweep needs different composition from a 6-panel comparison.** `compose.grid` captions every cell; repeated 64 times, two numbers per cell bury the shapes they label. The new `compose.matrix` puts the values on the axes instead, and sizes its fonts from the panel width rather than in absolute pixels — a sweep composes to ~6,000 px before `save()` downscales to 2,000, so a fixed 30 px label reaches the reader under 10 px.

## Related

The saturation noted under trade-offs is the same effect described in #43: the surface term stops improving long before the volumetric term does. Nothing here implements a better Chamfer, so that issue stays open.

**Comment by KhoiDOO** [comment]
Implemented and covered by PR #47.
